hbase-balancer/src/main/java/org/apache/hadoop/hbase/master/balancer/BaseLoadBalancer.java

```diff
@@ -617,4 +617,14 @@ private void roundRobinAssignment(BalancerClusterState cluster, List<RegionInfo>
   public void onConfigurationChange(Configuration conf) {
     loadConf(conf);
   }
+
+  @Override
+  public void reloadConfiguration() {
+    LOG.info("Reloading balancer configs");
+    // We clone and reload the clone because the original conf may be used in other
+    // non-balancer contexts. We only want to reload configs for the balancer.
+    Configuration conf = new Configuration(getConf());
```

Comment:
The reason I did this is I felt like it was even more confusing to developers if a Balancer-specific reloadConfigs option caused the entire HMaster to reload all of its configs. I think the expectation for such a feature would be that it only reloads the balancer-related configs.

One thing I could do is rename isReloadConfigs to isReloadMasterConfigs or something. This would make it clearer that the operation will reload everything.

What do you think of the trade off?

Reply:
I think it is fine to just reload the balancer's config, the concern is about the place.